File: python-backend/app/models/author.py
"""Author ORM model."""

from sqlalchemy import BigInteger, Boolean, ForeignKey, Integer, String, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class Author(Base):
    __tablename__ = "authors"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    repo_id: Mapped[str] = mapped_column(String, ForeignKey("repositories.id", ondelete="CASCADE"), nullable=False)
    name: Mapped[str] = mapped_column(String, nullable=False)
    email: Mapped[str] = mapped_column(String, nullable=False)
    is_default: Mapped[bool] = mapped_column(Boolean, default=False)
    created_at: Mapped[int] = mapped_column(BigInteger, nullable=False)
    updated_at: Mapped[int] = mapped_column(BigInteger, nullable=False)

    repository = relationship("Repository", back_populates="authors")

    __table_args__ = (
        UniqueConstraint("repo_id", "email", name="authors_repo_id_email_key"),
    )

File: golang-backend/internal/model/author.go
package model

import "gorm.io/gorm"

type Author struct {
	ID        int64  `gorm:"primaryKey;autoIncrement" json:"id"`
	RepoID    string `gorm:"column:repo_id;not null;uniqueIndex:idx_repo_email" json:"repoId"`
	Name      string `gorm:"not null" json:"name"`
	Email     string `gorm:"not null;uniqueIndex:idx_repo_email" json:"email"`
	IsDefault bool   `gorm:"column:is_default;default:false" json:"isDefault"`
	CreatedAt int64  `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt int64  `gorm:"column:updated_at" json:"updatedAt"`

	Repository Repository `gorm:"foreignKey:RepoID;references:ID;onDelete:CASCADE" json:"-"`
}

func (Author) TableName() string { return "authors" }

func (a *Author) BeforeCreate(tx *gorm.DB) error {
	if a.CreatedAt == 0 {
		a.CreatedAt = NowMs()
	}
	if a.UpdatedAt == 0 {
		a.UpdatedAt = NowMs()
	}
	return nil
}
